File: WebSocketMVVM/app/src/main/java/com/wildanka/websocketmvvm/view/MainActivity.java
package com.wildanka.websocketmvvm.view;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.wildanka.websocketmvvm.R;
import com.wildanka.websocketmvvm.model.entity.MarketData;
import com.wildanka.websocketmvvm.viewmodel.MainActivityViewModel;

public class MainActivity extends AppCompatActivity {
    private MainActivityViewModel viewModel;
    private static final String TAG = "MainActivity";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final TextView tvTest = findViewById(R.id.tv_last_price);

        viewModel = ViewModelProviders.of(this).get(MainActivityViewModel.class);
        viewModel.getMarketDataMutableLiveData("get update data btcidr").observe(this, new Observer<MarketData>() {
            @Override
            public void onChanged(@Nullable MarketData marketData) {
                Log.d(TAG, " last price : "+marketData.getLastPrice()+" \n price change : "+marketData.getPriceChange());
                tvTest.setText(marketData.getLastPrice());
            }
        });


    }
}

File: PagingStackexchange/app/src/main/java/com/wildanka/pagingstackexchange/MainActivity.kt
package com.wildanka.pagingstackexchange

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.wildanka.pagingstackexchange.model.entity.Items

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)

        viewModel.fetchStackAnswer().observe(this, Observer<List<Items?>>{ response ->
            if (response != null) {
                Toast.makeText(this@MainActivity, response.size.toString(), Toast.LENGTH_LONG ).show()
            }
        })
    }
}
